xncml/parser.py

```diff
@@ -0,0 +1,584 @@
+    ref : xr.Dataset
+      Reference dataset used to copy content into `target`.
+    obj : Group | Netcdf
+       <netcdf> object description.
+
+    Returns
+    -------
+    xr.Dataset
+      Dataset holding variables and attributes defined in <netcdf> element.
+    """
+    dims = {}
+    for item in obj.choice:
+        match item:
+            case Dimension():
+                dims[item.name] = read_dimension(item)
+            case Variable():
+                target = read_variable(target, ref, item, dims)
+            case Attribute():
+                read_attribute(target, item, ref)
+            case Remove():
+                target = read_remove(target, item)
+            case EnumTypedef():
+                raise NotImplementedError
+            case Group():
+                target = read_group(target, ref, item)
+            case Aggregation():
+                pass  # <aggregation> elements are parsed in `read_netcdf`
+            case _:
+                raise AttributeError
+
+    return target
+
+
+def read_scan(obj: Aggregation.Scan, ncml: Path) -> [xr.Dataset]:
+    """
+    Return list of datasets defined in <scan> element.
+
+    Parameters
+    ----------
+    obj : Aggregation.Scan instance
+      <scan> object description.
+    ncml : Path
+      Path to NcML document, sometimes required to follow relative links.
+
+    Returns
+    -------
+    list
+      List of datasets found by scan.
+    """
+    import re
+
+    if obj.date_format_mark:
+        raise NotImplementedError
+
+    path = Path(obj.location)
+    if not path.is_absolute():
+        path = ncml.parent / path
+
+    files = path.rglob("*") if obj.subdirs else path.glob("*")
+
+    if obj.reg_exp:
+        pat = re.compile(obj.reg_exp)
+        files = filter(pat.match, map(str, files))
+    elif obj.suffix:
+        pat = "*" + obj.suffix
+        files = [f for f in files if f.match(pat)]
+
+    files.sort()
+
+    return [xr.open_dataset(f, decode_times=False) for f in files]
+
+
+def read_coord_value(nc: Netcdf, agg: Aggregation, dtypes: list = ()):
+    """
+    Read `coordValue` attribute of <netcdf> element.
+
+    Parameters
+    ----------
+    nc : Netcdf instance
+      <netcdf> object description.
+    agg : Aggregation instance
+      <aggregation> object description
+    dtypes : tuple
+      List of preferred type for coordinate value.
+
+    Returns
+    -------
+    str, np.array, scalar
+      Coordinate value cast to proper type.
+
+    Notes
+    -----
+    The casting logic is most likely not up to spec.
+    """
+    val = nc.coord_value
+
+    match agg.type:
+        # A JOIN_NEW aggregation has exactly one coordinate value
+        case AggregationType.JOIN_NEW:
+            coord = val
+        case AggregationType.JOIN_EXISTING:
+            if "," in val:
+                coord = val.split(",")
+            elif " " in val:
+                coord = val.split(" ")
+            else:
+                coord = [val]
```

Comment:
The annotated schema linked above says : "blanks **and/or** commas", so you might want to do:
```suggestion
            coord = val.replace(',', ' ').split()
```
Split with no arguments catches all whitespaceS and line breaks and returns a list even if there is a single value.

I think my suggestion is slightly less explicit thant yours, but it think catches more corner cases.